djangoproject/djangoapp/permissions.py

```diff
@@ -0,0 +1,12 @@
+from rest_framework.permissions import IsAdminUser
+
+class IsAdminUserForRetrieve(IsAdminUser):
```

Comment:
Kao sto vidis vec psotoji permission IsAdminUser i on radi bas tu provjeru da li je admin korisnik https://www.django-rest-framework.org/api-guide/permissions/#isadminuser . Sto znaci da ovo nam nije potreban, isto tako nikad neces praviti permisiju na ovaj nacin, odnosno unutar permisije da se pitas ako je neka vrsta akcije pa valdiiras onda, jer same permisije ne treba da vode racuna o tome, znaci njihova duznost je da ispita da li je nesto validno i da li korsinik ima pristup necemu, ne treba da vodi racuna o tome da li je pozvano na `retrieve`, `save` ili nesto trece. Sto se tice koriscenja permisije za odredjenu rutu to mozes ovako odraditi https://www.django-rest-framework.org/api-guide/viewsets/#introspecting-viewset-actions

Reply:
Ovo ima mnogo vise smisla, ispravio sam da se za `retrieve` putanju postavlja permisija unutar viewset-a.